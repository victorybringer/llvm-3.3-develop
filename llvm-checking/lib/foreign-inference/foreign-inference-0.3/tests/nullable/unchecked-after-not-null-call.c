#include "base.h"

/*

  Here, arr is passed to a call that checks it for NULL.  This should
  not affect the not-nullability in the caller (f) since it could
  *still* be null.

 */

int f(int * arr, int * arr2) {
  int x = argTwoNotNull(arr, arr2);

  return x + arr[5];
}
