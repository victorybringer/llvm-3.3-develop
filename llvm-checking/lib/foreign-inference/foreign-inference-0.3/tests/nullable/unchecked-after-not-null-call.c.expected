fromList [ ("@f", fromList ["%arr", "%arr2"]) ]
